src/components/structures/MessagePanel.js

```diff
@@ -286,6 +287,47 @@ module.exports = React.createClass({
 
             var last = (i == lastShownEventIndex);
 
+            var isMembershipChange = (e) =>
+                e.getType() === 'm.room.member'
+                && ['join', 'leave'].indexOf(e.event.content.membership) !== -1
+                && (!e.event.prev_content || e.event.content.membership  !== e.event.prev_content.membership);
+
+            // Wrap consecutive member events in a ListSummary
+            if (isMembershipChange(mxEv)) {
+                let summarisedEvents = [mxEv];
+                i++;
+                for (;i < this.props.events.length; i++) {
+                    let collapsedMxEv = this.props.events[i];
+
+                    if (!isMembershipChange(collapsedMxEv)) {
+                        i--;
+                        break;
+                    }
+                    summarisedEvents.push(collapsedMxEv);
+                }
+                let renderEvents = (pEvent, events) => {
```

Comment:
let's have a comment here to say what we expect `i` to be. Is it the last in the group, or the first that's not in the group?